modules/ztp-using-pgt-to-maximize-power-saving-mode.adoc

```diff
@@ -0,0 +1,146 @@
+
+:_module-type: PROCEDURE
+[id="ztp-using-pgt-to-maximize-power-savings-mode_{context}"]
+= Maximizing power savings
+
+PolicyGenTemplate custom resources (CRs) allow you to overlay additional configuration details on top of the base source CRs provided with the GitOps plugin in the `ztp-site-generate` container.
+
+Capping the maximum CPU frequency is recommended to achieve maximum power savings.
+Enabling C-states on the non-critical workload CPUs without restricting the maximum CPU frequency negates much of the power savings by boosting the frequency of the critical CPUs.
+
+In "Using PolicyGenTemplate CRs to configure power savings mode" you set the power savings mode.
+
+Maximize power savings by updating the `sysfs` plugin fields setting an appropriate value for `max_perf_pct` in the TunedPerformancePatch CR for the reference configuration. This example based on the `group-du-sno-ranGen.yaml` describes the procedure to follow to restrict the maximum CPU frequency.
+
+You can find out the maximum supported frequency for each CPU core by running the command `cat /sys/devices/system/cpu/cpu*/cpufreq/cpuinfo_max_freq` on the cluster nodes.  The value is in kHz.
+
+As a starting point for `max_perf_pct`, you can use a percentage that caps all CPUs at the `All Cores Turbo frequency`.
+
+To calculate the `All Cores Turbo frequency` you need to find the minimum value between `CPU max MHz` and `cpuinfo_max_freq` for all CPU cores. This is because the `All Cores Turbo frequency` is the maximum frequency that all CPU cores can sustain simultaneously when fully occupied.
+
+Find the `CPU max MHz` by running the command `lscpu | grep "CPU max MHz"`. For example, if `CPU max MHz` is 4.0 GHz and `cpuinfo_max_freq` is 3.8 GHz for all CPU cores, then the `All Cores Turbo frequency` would be 3.8 GHz.
+
+.Prerequisites
+
+* You have configured power savings mode as described in "Using PolicyGenTemplate CRs to configure power savings mode".
+
+.Procedeure
+
+
+. Update the `PolicyGenTemplate` entry for `TunedPerformancePatch` in the `group-du-sno-ranGen.yaml` reference file in `out/argocd/example/policygentemplates` adding `max_perf_pct` as follows to maximize power savings.
++
+[source,yaml]
+----
+- fileName: TunedPerformancePatch.yaml
+      policyName: "config-policy"
+      spec:
+        profile:
+          - name: performance-patch
+            # The cmdline_crash CPU set must match the 'isolated' set in the PerformanceProfile above
+            data: |
+              [main]
+              summary=Configuration changes profile inherited from performance created tuned
+              include=openshift-node-performance-openshift-node-performance-profile
+              [bootloader]
+              cmdline_crash=nohz_full=2-19,22-39
+              [sysctl]
+              kernel.timer_migration=1
+              - fileName: TunedPerformancePatch.yaml
+      policyName: "config-policy"
```

Comment:
It looks like part of this YAML is accidentally duplicated. There should only be one `policyName` and accompanying `spec`. The second one looks correct

Reply:
changes lgtm